src/compile/data/colorrank.ts

```diff
@@ -19,6 +19,7 @@ export namespace colorRank {
    * Return hash dict from a color field's name to the sort and rank transforms
    */
   export function parseUnit(model: Model) {
+    // FIXME: why is this an array?
     let colorRankComponent: Dict<VgTransform[]> = {};
     if (model.has(COLOR) && model.fieldDef(COLOR).type === ORDINAL) {
```

Comment:
Note: Actually, if we include color rank as a part of the main transform, it's causing side effect to the data order.
I assume that this is fine because we never guarantee any particular data order.  (And if users want any specific order, 
they should use the `order` channel anyway.)  So this should be fine. 

(If we consider this problematic, this was a problem with my old code as well.)